subsys/net/lib/sockets/CMakeLists.txt

```diff
@@ -17,13 +17,10 @@ endif()
 
 zephyr_sources_ifdef(CONFIG_NET_SOCKETS_CAN                sockets_can.c)
 zephyr_sources_ifdef(CONFIG_NET_SOCKETS_PACKET             sockets_packet.c)
+zephyr_sources_ifdef(CONFIG_NET_SOCKETS_SOCKOPT_TLS        sockets_tls.c)
```

Comment:
I think we need an option to disable native TLS unrelated to `CONFIG_NET_SOCKETS_SOCKOPT_TLS` option, so it doesn't occupy space when offloaded TLS is supposed to be used only. But that can be done in separate PR, just expressing my thoughts.

Reply:
Why do you think we need another option for TLS? From my point of view, `CONFIG_NET_SOCKETS_SOCKOPT_TLS` doesn't need to be used by offloading, just as CAN/PACKET arent used by it, so it can be used exclusively to enable native TLS support. The name is quite unfortunate though, IMO we should drop the `SOCKOPT` part...

Reply:
>The name is quite unfortunate though, IMO we should drop the SOCKOPT part...

Indeed. It is hard to tell what does that option enable or what it SHOULD enable. Is it native TLS (which it does now), TLS credentials storage or rather TLS extensions to `setsockopt()` (which the name suggest, but it isn't the case I think).